NeuroID/NIDCallStatusObserver.swift

```diff
@@ -38,8 +38,14 @@ class NIDCallStatusObserver: NSObject, CXCallObserverDelegate {
     
     func startListeningToCallStatus(){
         if(!isRegistered){
-            self.callObserver.setDelegate(self, queue: nil)
-            isRegistered = true
+            _ = NIDConfigService { success in
```

Comment:
Why do we need to completely reinitialize this config class? Since we construct the call listener class in the config after we create the config class instance can we not dependency inject it instead and use the values?

Reply:
Alternatively rather than put any sort of depends in the class we can just check in the NeuroID class where we call these functions using the config value you there. That would keep this class totally independent of the config